src/main/webapp/app/pages/companionDiagnosticDevicesPage/companionDiagnosticDevicePage.tsx

```diff
@@ -0,0 +1,464 @@
+import OncoKBTable, {
+  SearchColumn,
+} from 'app/components/oncokbTable/OncoKBTable';
+import Select from 'react-select';
+import companionDiagnosticDevices from 'content/files/companionDiagnosticDevices/companion_diagnostic_devices.json';
+import { COMPONENT_PADDING, LG_TABLE_FIXED_HEIGHT } from 'app/config/constants';
+import { filterByKeyword } from 'app/shared/utils/Utils';
+import React, { useEffect, useState, useMemo } from 'react';
+import { Button, Col, Row } from 'react-bootstrap';
+import _ from 'lodash';
+import classnames from 'classnames';
+import { Input } from 'reactstrap';
+import WithSeparator from 'react-with-separator';
+import { AlterationPageLink, GenePageLink } from 'app/shared/utils/UrlUtils';
+import { pluralize } from 'cbioportal-frontend-commons';
+import { downloadFile } from 'app/shared/utils/FileUtils';
+
+export interface ICompanionDiagnosticDevice {
+  name: string;
+  manufacturer: string;
+  platformType?: string | null;
+  biomarkerAssociation: IBiomarkerAssociation;
+  specimenTypes?: string[];
+}
+
+interface IBiomarkerAssociation {
+  gene: string;
+  alterations: string[];
+  drugs: string[];
+  cancerType: string;
+}
```

Comment:
Once we start using the rc branch from transcript, these models should be auto generated.